features/navigator.yml.dist

```diff
@@ -0,0 +1,104 @@
+# Generated from: navigator.yml
+# Do not edit this file by hand. Edit the source file instead!
+
+status:
+  baseline: high
+  baseline_low_date: 2015-07-29
+  baseline_high_date: 2018-01-29
+  support:
+    chrome: "1"
+    chrome_android: "18"
+    edge: "12"
+    firefox: "1"
+    firefox_android: "4"
+    safari: "1"
+    safari_ios: "1"
+compat_features:
+  # ⬇️ Same status as overall feature ⬇️
+  # baseline: high
+  # baseline_low_date: 2015-07-29
+  # baseline_high_date: 2018-01-29
+  # support:
+  #   chrome: "1"
+  #   chrome_android: "18"
+  #   edge: "12"
+  #   firefox: "1"
+  #   firefox_android: "4"
+  #   safari: "1"
+  #   safari_ios: "1"
+  - api.Navigator
+  - api.Navigator.language
+  - api.Navigator.userAgent
+
+  # baseline: high
+  # baseline_low_date: 2015-07-29
+  # baseline_high_date: 2018-01-29
+  # support:
+  #   chrome: "2"
+  #   chrome_android: "18"
+  #   edge: "12"
+  #   firefox: "1.5"
+  #   firefox_android: "4"
+  #   safari: "4"
+  #   safari_ios: "3.2"
+  - api.Navigator.onLine
+
+  # baseline: high
+  # baseline_low_date: 2017-10-17
+  # baseline_high_date: 2020-04-17
+  # support:
+  #   chrome: "37"
+  #   chrome_android: "37"
+  #   edge: "16"
+  #   firefox: "32"
+  #   firefox_android: "32"
+  #   safari: "10.1"
+  #   safari_ios: "10.3"
+  - api.Navigator.languages
+
+  # baseline: low
+  # baseline_low_date: 2023-03-27
+  # support:
+  #   chrome: "94"
+  #   chrome_android: "94"
+  #   edge: "94"
+  #   firefox: "99"
+  #   firefox_android: "99"
+  #   safari: "16.4"
+  #   safari_ios: "16.4"
+  - api.Navigator.pdfViewerEnabled
+
+  # baseline: low
+  # baseline_low_date: 2024-09-16
+  # support:
+  #   chrome: "1"
+  #   chrome_android: "18"
+  #   edge: "12"
+  #   firefox: "1"
+  #   firefox_android: "4"
+  #   safari: "18"
+  #   safari_ios: "18"
+  - api.Navigator.cookieEnabled
+
+  # baseline: false
+  # support:
+  #   chrome: "37"
+  #   chrome_android: "37"
+  #   edge: "15"
+  #   firefox: "48"
+  #   firefox_android: "48"
+  - api.Navigator.hardwareConcurrency
```

Comment:
Hrm, I meant to bring this up in the original PR comment and forgot.

Caniuse has features for even more APIs that hang off here, eg `onLine`, `registerProtocolHandler`.

I was considering breaking them all out as features and adding a new group for them all, but wanted to get your take on it first before featurizing all of these bits, since it also leaves `navigator` oddly empty and "misc" feeling.

Reply:
It's true that `navigator` (also true for `window`) is mostly a global object that lets you access other features.

- `cookieEnabled` could go into a cookie feature, especially important once we start defining a feature for partitioned cookies.
- `onLine` might also be interesting as its own feature, together with the `online` and `offline` event listeners.
- I would definitely also want to split out the `registerProtocolHandler` things in their own feature. And probably group them with `html.manifest.protocol_handlers`.
- `userAgent` might also be interesting to consider on its own, especially because it would give us a way to add a description like "Not to be used anymore because UA Strings lie, and are hard to parse. Use client hints instead".
- ...

I'm having a real hard time finding properties and methods of the `navigator` global object that would make sense to keep here instead of in their own features.
I'm not completely opposed to having a navigator feature that acts as a catch-all for very old things no one cares about anymore, but a feature like this would be hard to name and describe too.

Other options include:
- Ignore `api.Navigator` altogether. It's a given, it will never not exist, and isn't, in itself, a feature.
- Or create a new feature called something like `global-browser-objects` which contains `api.Navigator`, `api.Window`, and other similar objects like `self`, `frames`, `globalThis`...

Reply:
Ok thanks, I'm flipping this back to draft, and will break out these as features. I agree that most of them stand alone.

>     * Ignore `api.Navigator` altogether. It's a given, it will never not exist, and isn't, in itself, a feature.

Hm, I am not a fan of leaving significantly visible parts of the platform out. That would just make things harder for newcomers. I'd prefer a symbolic placeholder feature w/ just that one single key, over leaving it out. I'll take that approach for now.

>     * Or create a new feature called something like `global-browser-objects` which contains `api.Navigator`, `api.Window`, and other similar objects like `self`, `frames`, `globalThis`...

I'm going w/ a mostly-empty Navigator for now. Maybe in the next meeting we could discuss an approach which brings these all together, whether feature or group.

Reply:
Drafted this approach. Descriptions need more work, but here are some notes on feature composition approach. I don't think there's anything that's hard blocking here, but lmk if have suggestions for fixes or different approaches.

- `navigator` as a standalone feature... optionally we could make a group for all of the APIs hanging off these mega-objects, but if so we should do that consistently
- `navigator.cookiesEnabled` as a feature for now, because likely we end up w/ a group of cookie features due to implementation timeline and feature distinction
- `navigator.hardwareConcurrency` is currently in conflict because Caniuse has it all green w/ a note about Safari having it behind a flag, whereas BCD has it as baseline false due to no Safari support (i added a comment for now)
- `navigator.language` and related keys in a `Language` feature, but need to figure out whether to include text/audio/video track `language` properties, and also whether to include in the `intl` group (wondering how l10n fits there too...)
- `navigator.onLine` and friends - Doesn't quite match up exactly with Caniuse, since BCD reports supported while Caninuse considers the early Chrome/Safari partial implementation as not supported, just added a comment for now
- `navigator.pdfViewerEnabled` is all green on Caniuse but they have a separate feature for PDF viewing generally, which is marked not supported on Firefox/Chrome Android - i made a note on this feature, don't think we need to block on finding a magical fix
- `navigator.registerProtocolHandler` has a bunch of subkeys on BCD for supported schemes, etc - should figure out whether to include or not
- `navigator.userAgent` in a feature called `Client identification (user agent)` which is named to align w/ the spec and the direction towards client hinting over UA string